File: lessons/compiled_lang/rust/src/m00.function.rs
fn add_nums(numbers: &[i32]) -> i32 {
    let mut sum = 0;
    for num in numbers {
        sum += num;
    }
    sum
}

fn main() {
    println!("The numbers to be added are 5, 2, 4, 3, 6.");

    let result = add_nums(&[5, 2, 4, 3, 6]);
    println!("The result of their summation is: {result}.");
}

File: lessons/compiled_lang/java/src/m00.function.java
// Not public - see a00.output.java for why.
class M00Function {
    static int addNums(int... numbers) {
        int sum = 0;
        for (int num : numbers) {
            sum += num;
        }
        return sum;
    }

    public static void main(String[] args) {
        System.out.println("The numbers to be added are 5, 2, 4, 3, 6.");

        int result = addNums(5, 2, 4, 3, 6);
        System.out.println("The result of their summation is: " + result + ".");
    }
}
